hknweb/events/views.py

```diff
@@ -1,18 +1,82 @@
 from django.shortcuts import render, redirect
 from django.http import HttpResponse
 from django.template import loader
+from django.contrib import messages
+from django.views.decorators.csrf import csrf_exempt #doing this for now bc idk how to make csrf work
 
-from .models import Event
+from .models import Event, Rsvp
+from hknweb.models import Profile
 
 def index(request):
-    events = Event.objects.order_by('-created_at')[:10]
-
+    events = Event.objects.order_by('-start_time')[:10]
     context = {
         'events': events,
     }
     return render(request, 'events/index.html', context)
 
 
+def show_details(request, id):
+    event = Event.objects.get(pk=id)
+    try:
+        rsvp = Rsvp.objects.get(user=Profile.objects.get(user=request.user), event=event)
+        rsvp = True
+    except Rsvp.DoesNotExist:
+        rsvp = False
+
+    context = {
+        'event': event,
+        'rsvp': rsvp
+    }
+    return render(request, 'events/show_details.html', context)
+
+@csrf_exempt  #doing this for now bc idk how to make csrf work
+def rsvp(request, id):
```

Comment:
Apparently there's a `@login_required` decorator you can use for this? 
```py
from django.contrib.auth.decorators import login_required

@login_required
def my_view(request):
    ...
```

https://docs.djangoproject.com/en/2.1/topics/auth/default/#the-login-required-decorator